src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -4490,7 +4495,16 @@ FindReferencedTableColumn(Expr *columnExpression, List *parentQueryList, Query *
 		return;
 	}
 
-	Index rangeTableEntryIndex = candidateColumn->varno - 1;
+	if (candidateColumn->varattno == 0)
```

Comment:
These changes are needed because `GroupedByPartitionColumn` now descends into this function for subquery pushdown MultiTable nodes, which may not have a partition column and may have odd target list entries.